### DISCUSSION TITLE
Treatment notes display bug

### DISCUSSION THREAD

**Discussion by Grandrux** [Navid200/xDrip#58] [General]
When adding a note to existing carb or insulin treatment it becomes "stuck" to a grid, so when changing glucose units it moves accordingly to a number it was at. Here's what it looks like:

Here you can see 2 treatments: 1U of Actrapid at 20:00  and 1g of carbs at 20:15. GU are in mmol/l

![Screenshot_2021-08-09-20-16-22-725_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128754491-2542902a-ec12-4dea-99d6-142cec5c7a09.jpg)

![Screenshot_2021-08-09-20-16-26-324_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128756155-7aaf3a0d-c95f-41bd-98e0-34210cf4c36f.jpg)

When GU are changed to mg/dl they stay in place

![Screenshot_2021-08-09-20-16-42-611_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128756596-7725ea2f-c268-4a1c-9219-c1ea65080e26.jpg)

![Screenshot_2021-08-09-20-16-46-327_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128756618-88941f7a-ee98-45f0-8b97-fc2f989fc268.jpg)

Now I add  a "test" note to insulin treatment while the GU are in mmol/l

![Screenshot_2021-08-09-20-17-24-001_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128757263-db8828c3-1d01-420a-ba44-1a6a05cc258d.jpg)

and after changing to mg/dl this is what happens

![Screenshot_2021-08-09-20-17-39-907_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128757580-6630686e-651e-45df-a824-1af51b79eee0.jpg)

the treatment dot became fixed at 3.9 mmol/l, and it moved to the bottom of the graph since the grid values are now in hundreds.

Same thing happens to a carb treatment

![Screenshot_2021-08-09-20-18-20-514_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128758923-048a4a36-6f02-41e6-95ec-3c68b04e7cfa.jpg)

Now the opposite happens when the note was added while the GU are in mg/dl.
Here, while in mg/dl, you can see insulin treatment with a note that was added while being in mmol/l (so its at the bottom), and a "test" note added to 1g of carbs. 

![Screenshot_2021-08-09-20-20-30-009_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128762134-6da2f733-c4af-4f77-b8c2-97b4d8849764.jpg)

When the units are changed to mmol/l it becomes  like this

![Screenshot_2021-08-09-20-20-43-986_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/81015515/128760747-a96cd05f-00a0-4fa1-84f6-c8a39fcfb456.jpg)

the insulin treatment dot now sits at  the top of the red area at 3.9 mmol/l, and the carb treatment dot is now at the top of the screen sitting on 126 mg/dl

The same thing happens to the treatment values that are in history and have notes  attached to them. Values without added notes don't change their position.

All this was done on the latest nightly build available at the moment of this writing, which is e4f7fa3-2021.08.04, but the same thing happened on previous builds so the build is not the issue here.

**Comment by Navid200** [comment]
Sorry about the delay.
I believe this is related to an open issue:
https://github.com/NightscoutFoundation/xDrip/issues/1299

I have an open PR for it, which I believe will address this as well.
https://github.com/NightscoutFoundation/xDrip/pull/1811
